synapse/app/homeserver.py

```diff
@@ -406,6 +411,24 @@ def phone_stats_home():
         daily_messages = yield hs.get_datastore().count_daily_messages()
         if daily_messages is not None:
             stats["daily_messages"] = daily_messages
+        else:
+            stats.pop("daily_messages", None)
+
+        if first_time:
+            # Report the stats as synapse metrics.
+            # As some of the stats are expensive to calculate we only update
+            # them when synapse phones home to matrix.org every 24 hours.
```

Comment:
Can you comment that we only want to *set* the callbacks, as its very confusing since this function does only get called once every 24 hours.